static/src/javascripts/projects/common/modules/navigation/search.js

```diff
@@ -0,0 +1,169 @@
+            config.page.googleSearchUrl &&
+            config.page.googleSearchId &&
+            toggle
+        ) {
+            this.gcsUrl = `${config.page.googleSearchUrl}?cx=${config.page
+                .googleSearchId}`;
+            this.resultSetSize = config.page.section === 'identity' ? 3 : 10;
+
+            searchLoader = throttle(() => {
+                this.load();
+            });
+
+            bean.on(toggle, 'click', e => {
+                const popup = document.querySelector('.js-search-popup');
+                const maybeDismissSearchPopup = event => {
+                    let el = event.target;
+                    let clickedPop = false;
+
+                    if (popup) {
+                        while (el && !clickedPop) {
+                            /* either the search pop-up or the autocomplete resultSetSize
+                               NOTE: it would be better to check for `.gssb_c`,
+                                     which is the outer autocomplete element, but
+                                     google stops the event bubbling earlier
+                            */
+                            if (
+                                el &&
+                                el.classList &&
+                                (el.classList.contains('js-search-popup') ||
+                                    el.classList.contains('gsq_a'))
+                            ) {
+                                clickedPop = true;
+                            }
+
+                            el = el.parentNode;
+                        }
+
+                        if (!clickedPop) {
+                            event.preventDefault();
+                            toggle.classList.remove('is-active');
+                            popup.classList.add('is-off');
+                            bean.off(
+                                document,
+                                'click',
+                                maybeDismissSearchPopup
+                            );
+                        }
+                    }
+                };
+
+                setTimeout(() => {
+                    if (toggle.classList.contains('is-active')) {
+                        bean.on(document, 'click', maybeDismissSearchPopup);
+                    }
+                });
+
+                searchLoader();
+                // Make sure search is always in the correct state
+                focusSearchField();
+                e.preventDefault();
+            });
+        }
+    }
+
+    load(): void {
+        let s;
+        let x;
+
+        const container = document.querySelector('.js-search-placeholder');
+
+        // Set so Google know what to do
+        // eslint-disable-next-line no-underscore-dangle
+        window.__gcse = {
+            callback: focusSearchField,
+        };
+
+        // Unload any search placeholders elsewhere in the DOM
+        Array.prototype.forEach.call(
```

Comment:
You could do this here `[...document.querySelectorAll('.js-search-placeholder')].forEach`